apollo/lib/websocket_manager.py

```diff
@@ -0,0 +1,39 @@
+import uuid
+
+from fastapi import WebSocket
+from starlette.websockets import WebSocketDisconnect
+from typing import Dict
+
+from apollo.lib.singleton import Singleton
+
+
+class WebSocketManager(metaclass=Singleton):
+    def __init__(self):
+        self.connections: Dict[uuid.UUID, WebSocket] = {}
+
+    async def add_and_connect_websocket(self, websocket: WebSocket):
+        self.connections[uuid.uuid4()] = websocket
+
+        await websocket.accept()
+        await websocket.send_json("Connection accepted")
+        try:
+            return await websocket.receive_json()
+        except WebSocketDisconnect:
+            return
+
+    async def send_message(self, websocket_id: uuid.UUID, message: str):
```

Comment:
I foresee some race conditions if you were to actually call `send_message` async like if you'd for example do `cd` and then `ls`. `ls` _could_ be sent earlier.

Part of it is also shell session management in a context with multiple users but that's out of scope. The only concern I have regarding this that's in scope is that it'd be nice to not lay the responsibility of guaranteeing a sync call on the callee.